tests/src/app/testqgsmaptoolcircularstring.cpp

```diff
@@ -104,6 +105,34 @@ void TestQgsMapToolCircularString::testAddCircularStringCurvePoint()
   QgsSettings().setValue( QStringLiteral( "/qgis/digitizing/default_z_value" ), 0 );
 }
 
+void TestQgsMapToolCircularString::testAddCircularStringCurvePointWithDeletedVertex()
+{
+  QgsSettings().setValue( QStringLiteral( "/qgis/digitizing/default_z_value" ), 333 );
+  mLayer->startEditing();
+
+  QgsMapToolCircularStringCurvePoint mapTool( mParentTool, mCanvas );
```

Comment:
What about for QgsMapToolAddCircularString too? The logic there is quite complex